x-pack/plugins/rule_registry/server/utils/create_persistence_rule_type_wrapper.ts

```diff
@@ -26,6 +26,10 @@ export const createPersistenceRuleTypeWrapper: CreatePersistenceRuleTypeWrapper
               const numAlerts = alerts.length;
               logger.debug(`Found ${numAlerts} alerts.`);
 
+              const ruleDataClientWriter = await ruleDataClient.getWriter({
+                namespace: options.spaceId,
+              });
+
```

Comment:
@marshallmain As discussed, this will fail the execution by throwing an error if writing is disabled (either via error or config) whether or not there are alerts to write.